src/consensus.js

```
@@ -266,6 +271,92 @@ let consensus = {
+                logr.error('Force finalize failed for ' + height + '#' + winner.block.hash.substr(0, 8), err)
+                consensus.possBlocks = consensus.possBlocks.filter(pb => pb.block.hash !== winner.block.hash)
+                candidates.shift()
+                if (candidates.length > 0 && retryCount < 3) {
+                    logr.warn('Trying next candidate for height ' + height + ' (retry ' + (retryCount + 1) + '/3)')
+                    consensus.finalizing = false
+                    setImmediate(() => consensus._forceFinalize(height, retryCount + 1))
+                    return
+                }
+                if (candidates.length > 0) {
+                    logr.fatal('Force finalize: retry limit exceeded for height ' + height + '. All candidates failed. Manual intervention required.')
+                }
```

ESLint

[error] Unnecessary { after 'if' condition.

(curly)